app/domain/operations/hbx_admin/dry_run/notice_query.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Operations
+  module HbxAdmin
+    module DryRun
+      # This Operation is responsible for building a query to fetch notice counts based on given parameters.
+      class NoticeQuery
+        include Dry::Monads[:do, :result]
+        include L10nHelper
+
+        # Calls the NoticeQuery operation.
+        #
+        # @param params [Hash] The parameters for the query.
+        # @option params [Array<String>] :person_hbx_ids The person HBX IDs.
+        # @option params [Date] :start_date The start date for the query.
+        # @option params [Date] :end_date The end date for the query.
+        # @option params [Array<String>] :title_codes The title codes for the query.
+        # @return [Dry::Monads::Result] The result of the operation.
+        def call(params)
+          validated_params = yield validate(params)
+          build_pipeline(validated_params)
```

Comment:
Make use of the yield and return a Moand from the method.

Reply:
I am just returning a hash, do we need a monad here?

Reply:
Yes, please. Let's be consistent with others.